lib/demos/reference/motion_demo_fade_scale_transition.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+// BEGIN fadeScaleTransitionDemo
+
+class FadeScaleTransitionDemo extends StatefulWidget {
+  const FadeScaleTransitionDemo();
+
+  @override
+  _FadeScaleTransitionDemoState createState() =>
+      _FadeScaleTransitionDemoState();
+}
+
+class _FadeScaleTransitionDemoState extends State<FadeScaleTransitionDemo>
+    with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    _controller = AnimationController(
+      value: 0,
+      duration: const Duration(milliseconds: 150),
+      reverseDuration: const Duration(milliseconds: 75),
+      vsync: this,
+    )..addStatusListener((status) {
+        setState(() {
+          // setState needs to be called to trigger a rebuild because
+          // the 'HIDE FAB'/'SHOW FAB' button needs to be updated based
+          // the latest value of [_controller.status].
+        });
+      });
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  bool get _isAnimationRunningForwardsOrComplete {
+    switch (_controller.status) {
+      case AnimationStatus.forward:
+      case AnimationStatus.completed:
+        return true;
+      case AnimationStatus.reverse:
+      case AnimationStatus.dismissed:
+        return false;
+    }
+    assert(false);
+    return null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+    final colorScheme = Theme.of(context).colorScheme;
+
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        title: Text(localizations.demoFadeScaleTitle),
+      ),
+      floatingActionButton: AnimatedBuilder(
+        animation: _controller,
+        builder: (context, child) {
+          return FadeScaleTransition(
+            animation: _controller,
+            child: child,
+          );
+        },
+        child: Visibility(
+          visible: _controller.status != AnimationStatus.dismissed,
+          child: FloatingActionButton(
+            child: const Icon(Icons.add),
+            onPressed: () {},
+          ),
+        ),
+      ),
+      bottomNavigationBar: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          const Divider(height: 0),
+          Padding(
+            padding: const EdgeInsets.symmetric(vertical: 8),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                RaisedButton(
+                  onPressed: () {
+                    showModal<void>(
+                      context: context,
+                      builder: (context) {
+                        return _ExampleAlertDialog();
+                      },
+                    );
+                  },
+                  color: colorScheme.primary,
+                  textColor: colorScheme.onPrimary,
```

Comment:
I think the button should already be using these colors because of the theme, so you can probably remove these 2 lines here and below

Reply:
will do